Libraries/Opc.Ua.Gds.Server.Common/CertificateGroup.cs

```diff
@@ -109,18 +142,25 @@ public virtual async Task Init()
                 store?.Close();
             }
 
-            if (Certificate == null)
+            foreach (KeyValuePair<NodeId, X509Certificate2> keyValuePair in Certificates)
             {
-                Utils.LogInfo(Utils.TraceMasks.Security,
-                    "Create new CA Certificate: {0}, KeySize: {1}, HashSize: {2}, LifeTime: {3} months",
-                    SubjectName,
-                    Configuration.CACertificateKeySize,
-                    Configuration.CACertificateHashSize,
-                    Configuration.CACertificateLifetime
-                    );
-                X509Certificate2 newCertificate = await CreateCACertificateAsync(SubjectName).ConfigureAwait(false);
-                Certificate = X509CertificateLoader.LoadCertificate(newCertificate.RawData);
-                Utils.LogCertificate(Utils.TraceMasks.Security, "Created CA certificate: ", Certificate);
+                X509Certificate2 certificate = keyValuePair.Value;
+                NodeId certificateType = keyValuePair.Key;
+
+                if (certificate == null)
+                {
+                    Utils.LogInfo(Utils.TraceMasks.Security,
+                        "Create new CA Certificate: {0}, CertificateType {1}  KeySize: {2}, HashSize: {3}, LifeTime: {4} months",
+                        SubjectName,
+                        certificateType,
+                        Configuration.CACertificateKeySize,
+                        Configuration.CACertificateHashSize,
+                        Configuration.CACertificateLifetime
+                        );
+                    X509Certificate2 newCertificate = await CreateCACertificateAsync(SubjectName, certificateType).ConfigureAwait(false);
+                    Utils.LogCertificate(Utils.TraceMasks.Security, "Created CA certificate: ", Certificates[certificateType]);
```

Comment:
Shouldn't we log newCertificate instead of Certificate[certificateType] which can be null in this case ?

Reply:
Ok, assumption is wrong since CreateCACertificateAsync both creates and adds it to Certificates[certificateType], but still newCertificate is never used ...